### DISCUSSION TITLE
OTBR: Accessing Border Agent ID from d-bus

### DISCUSSION THREAD

**Discussion by tleacmcsa** [openthread/openthread#10913] [Q&A] [answered]
When running otbr-agent on a linux platform, how can a client retrieve the border agent id which is being sent via the DNS-SD service record?  Matter's Thread Border Router Management Cluster needs this, for example.

If a mechanism doesn't exist today I assume a patch to introspect.xml and dbus_thread_object_rcp.cpp would be the right path?

**Comment by jwhui** [comment, marked as answer]
I believe the Border Agent ID is currently accessible today via the OpenThread API, CLI, or REST API. I don't think it is exposed via D-Bus yet.
